discussment-core/src/main/java/org/danekja/discussment/core/service/imp/DefaultPostService.java

```diff
@@ -107,6 +109,65 @@ public IDiscussionUser getPostAuthor(Post post) throws DiscussionUserNotFoundExc
         }
     }
 
+    public Post getLastPost(Discussion discussion) throws AccessDeniedException{
+        if(accessControlService.canViewPosts(discussion)) {
```

Comment:
This is terribly inefficient. Add method to postDao which return first post from posts by discussion ordered by created date in descending order.

Delete unused getLastPost methods from Post class.